### DISCUSSION TITLE
Documentation for Module creation

### DISCUSSION THREAD

**Discussion by HenrijsS** [nuxt/nuxt#18310] [Questions]
When's the ETA for Module documentation using the new nuxt-kit?
The NPM package doesn't link to any repo.

I remember there was a Private Beta period for Module creators but currently only 2 work.
To increase the pace, we should add more Collaborators and let others also know how to support the modules.

From what I've seen, the Strapi Module doesn't even use the new Nuxt-kit so I'd gladly help to convert it to the new version.
Or just have another set of developers maintain a separate version only for Nuxt 3.

**Comment by wobsoriano** [comment]
You can check this one as a basis for your module https://github.com/wobsoriano/nuxt3-supabase

If you don't need full `nuxt3` and only need `kit`, install this instead

`"@nuxt/kit": "npm:@nuxt/kit-edge@latest"`

then

```ts
import {
  defineNuxtModule,
  addPlugin,
  addTemplate
} from '@nuxt/kit'
```

**Reply by HenrijsS** [reply]
Yeah. I know about the Kit package, but it'd be good to wrap my head around the options it has. 
I've seen the defineNuxtModule, but a clear documentation would be superb.
That's what this dc was about.

**Reply by mirabledictu** [reply]
> You can check this one as a basis for your module https://github.com/wobsoriano/nuxt3-supabase
> 
> If you don't need full `nuxt3` and only need `kit`, install this instead
> 
> `"@nuxt/kit": "npm:@nuxt/kit-edge@latest"`
> 
> then
> 
> ```ts
> import {
>   defineNuxtModule,
>   addPlugin,
>   addTemplate
> } from '@nuxt/kit'
> ```

thanks for this. will use as a starter

**Reply by wobsoriano** [reply]
Yep same. I just checked the types and legacy docs to work on it

**Reply by HenrijsS** [reply]
Oh. Forgot about types 😁
Gonna check those rn.
